namer/core/src/main/scala/io/buoyant/namer/EnumeratingNamer.scala

```diff
@@ -0,0 +1,8 @@
+package io.buoyant.namer
+
+import com.twitter.finagle.{Name, NameTree, Path, Namer}
+import com.twitter.util.{Activity, Future}
+
+trait EnumeratingNamer extends Namer {
+  def getAllNames: Future[Set[Path]]
```

Comment:
should this expose an Activity as well? Polling for a set is a special case of watching, and I imagine it makes sense to watch this for many applications